### PR TITLE
Frame construction in onCompleted was breaking the browser keyboard focus

### PR DESCRIPTION
**Pull request by jherico**

### CODE REVIEW

File: interface/resources/qml/windows/Window.qml [hunk 1]
```diff
@@ -69,28 +69,9 @@ Fadable {
...
-        DefaultFrame { anchors.fill: parent }
-    }
-
-    Component {
-        id: modalFrameBuilder;
-        ModalFrame { anchors.fill: parent }
-    }
-
-    Component.onCompleted: {
-        if (!frame) {
```
Comment (jherico) [inline interface/resources/qml/windows/Window.qml]: The inability to properly focus the keyboard in a browser window was specifically caused by this code.  I suspect the reason is that the frame creation here doesn't establish the *required* Z-order between the frame, the content, and the activator `MouseArea`.  

To work around this, I've created a ModalWindow that defaults to the ModalFrame, for the same result without breaking the browser.

**Comment by ZappoMan** [comment]
@jherico - testing notes?

**Comment by hifi-gustavo** [comment]
Build console output for commit: 20be64020ac0b10a7b1a70af954389be8213de72 available at: https://gist.github.com/2fa113e0a5d6a6fe9f11<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR6886.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR6886.dmg

**Comment by murillodigital** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 20be64020ac0b10a7b1a70af954389be8213de72 available at: https://gist.github.com/2fa113e0a5d6a6fe9f11<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR6886.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR6886.dmg

**Comment by ZappoMan** [comment]
I think this PR is supposed to fix the focus issues with the Browser window... but it doesn't seem to fix things, and in fact, only makes them worse.

@jherico - can you please provide some testing notes, what's this PR do?

**Comment by hifi-gustavo** [comment]
Build console output for commit: 4f9fd3d14ddc1bef7ac5208543113273bbe85e6d available at: https://gist.github.com/2fa113e0a5d6a6fe9f11<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR6886.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR6886.dmg

**Comment by jherico** [comment]
Testing notes:

If you open a browser window (`Ctrl-B`) and go to a website that contains input fields you should now be able to type into the input fields.  

This may also apply to the Marketplace window

**Comment by hifi-gustavo** [comment]
Build console output for commit: f343598d9171aa745f1772b3ef9cc03fb1151e11 available at: https://gist.github.com/2fa113e0a5d6a6fe9f11<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR6886.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR6886.dmg

**Comment by thoys** [comment]
@jherico Tested this but can not type into input fields yet. Tested with page: https://highfidelity.com/login . The text doesn't get appended to the address bar anymore though.

**Comment by ZappoMan** [comment]
using the web installer work round, I have tested this and confirmed it works as expected. I am going to merge.

**Comment by thoys** [comment]
@ZappoMan webinstaller workaround? what did I miss

**Comment by jherico** [comment]
@thoys currently there is a bug affecting all web engine views in the production build.  To fix it you have to replace the `icudtl.dat` that is installed by the application with one from a local build of interface.  otherwise if you try to type into a web view it will crash the QtWebEngineProcess child process and the web view will vanish.  it's a build / installer bug.

**Comment by thoys** [comment]
ah thanks, will try that next time testing one of the web engine view PR's.

**Pull request merged**